### DISCUSSION TITLE
The monaco editor is not displayed properly in RTL

### DISCUSSION THREAD

**Discussion by sayyad84** [OrchardCMS/OrchardCore#11269] [Q&A]
Hi,
I can't use liquid editor (monaco) in rtl website to create templates. please help me to solve this problem.
when I change dir to ltr in div with id=Content_editor from browser's developer mode, I can type in text area but it shows them from right side.
thanks.

**Comment by hishamco** [comment]
Are using the nightly builds? If Yes let me know I will try to check this one

**Comment by sayyad84** [comment]
thank you for response.
I use latest stable version, 1.2.2.
I changed Create.cshtml in template module and now I can write in liquid textarea but text shows right to left.

**Reply by hishamco** [reply]
I will try to reproduce the issue in the code base

**Reply by hishamco** [reply]
Meanwhile if you have the source, please check coz I may have a look after few hours

**Comment by sayyad84** [comment]
I changed Create.cshtml and Edit.cshtml files in OrchardCore.Templates modules and set dir="ltr":
`<div id="@Html.IdFor(x => x.Content)_editor" asp-for="Text" style="min-height: 600px;" class="form-control" dir="ltr"></div>`
and I added text-align: left; to monaco-editor class in editor.main.css in OrchardCore.Resources module.
Problem solved.

**Reply by hishamco** [reply]
No need to add `dir` property, because the layout shoud set this in the TheAdmin theme

Regarding the monaco editor I'm expecting this done in the editor itself

**Comment by hishamco** [comment]
https://github.com/OrchardCMS/OrchardCore/blob/d6eab119e11046108ce37693b3c591c45884fe9a/src/OrchardCore.Modules/OrchardCore.Html/Views/HtmlBodyPart-Monaco.Edit.cshtml#L10-L15

The `dir` is already set in the monaco editor

**Reply by hishamco** [reply]
Are you sure that you enabled the localization module, and set the current culture to Arabic?

**Reply by sayyad84** [reply]
My culture is fa-Persian so direction will be "rtl" but this is not correct for monaco-editor and it must always be "ltr" with text-align: left;
I changed template module not html module.

**Comment by hishamco** [comment]
If you are using Persian this should works while the direction RTL

> I changed template module not html module.

Monaco editor is a part of HTML module

**Reply by jptissot** [reply]
@hishamco if you plan on modifying monaco, please let me know as I have a PR going for monaco improvements.

**Reply by hishamco** [reply]
I will check this ASAP, to make sure if there's an issue in RTL. So, I will let you know if there's a changes

**Reply by hishamco** [reply]
@jptissot seems there's a bug in Monaco editor itself, so I will create an issues based on this discussion

**Comment by sayyad84** [comment]
![1](https://user-images.githubusercontent.com/57654948/155876632-4ab77c32-7c04-4cf3-9ee3-2d4e4c6e304a.png)
![2](https://user-images.githubusercontent.com/57654948/155876638-0021906f-2808-4885-a175-b1b87c123eb0.png)
![3](https://user-images.githubusercontent.com/57654948/155876640-30abb8a4-e54b-4f89-b6b3-6ec174636ed1.png)
![4](https://user-images.githubusercontent.com/57654948/155876643-bdd6a382-813c-498e-b0ce-7f99f4de5dec.png)

**Reply by hishamco** [reply]
Propably there's an issue with the monaco lang assets

**Comment by hishamco** [comment]
I can confirm there's an issue with RTL, @sayyad84 are you sure you are able to edit the content, coz I'm unable to edit the content, also the data is not rendered when I switch to Arabic

**Reply by sayyad84** [reply]
Yes, after changes the template module and css file, there is no problem in liquid textarea in create template.

**Reply by hishamco** [reply]
The fix should be in the HTML module itself

**Comment by hishamco** [comment]
FYI https://github.com/microsoft/vscode-loc/issues/459#issuecomment-933977432
